internal/server/storage/drivers/driver_linstor_utils.go

```diff
@@ -0,0 +1,215 @@
+func (d *linstor) createResourceGroup() error {
+	d.logger.Debug("Creating Linstor resource group")
+
+	// Retrieve the Linstor client
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return fmt.Errorf("Could not load Linstor client: %w", err)
+	}
+
+	placeCount, err := strconv.Atoi(d.config[LinstorResourceGroupPlaceCountConfigKey])
+	if err != nil {
+		return fmt.Errorf("Could not parse resource group place count property: %w", err)
+	}
+
+	resourceGroup := linstorClient.ResourceGroup{
+		Name:        d.config[LinstorResourceGroupNameConfigKey],
+		Description: "Resource group managed by Incus",
+		SelectFilter: linstorClient.AutoSelectFilter{
+			PlaceCount: int32(placeCount),
+		},
+	}
+
+	if d.config[LinstorResourceGroupStoragePoolConfigKey] != "" {
+		resourceGroup.SelectFilter.StoragePool = d.config[LinstorResourceGroupStoragePoolConfigKey]
+	}
+
+	err = linstor.Client.ResourceGroups.Create(context.TODO(), resourceGroup)
+	if err != nil {
+		return fmt.Errorf("Could not create Linstor resource group : %w", err)
+	}
+
+	return nil
+}
+
+// deleteResourceGroup deleter the resource group for the storage pool.
+func (d *linstor) deleteResourceGroup() error {
+	// Retrieve the Linstor client
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return fmt.Errorf("Could not load Linstor client: %w", err)
+	}
+
+	err = linstor.Client.ResourceGroups.Delete(context.TODO(), d.config[LinstorResourceGroupNameConfigKey])
+	if err != nil {
+		return fmt.Errorf("Could not delete Linstor resource group : %w", err)
+	}
+
+	return nil
+}
+
+// getResourceDefinition returns a Resource Definition instance for the given Resource name.
+func (d *linstor) getResourceDefinition(resourceDefinitionName string) (*linstorClient.ResourceDefinition, error) {
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return nil, fmt.Errorf("Could not load Linstor client: %w", err)
+	}
+
+	resourceDefinition, err := linstor.Client.ResourceDefinitions.Get(context.TODO(), resourceDefinitionName)
+	if errors.Is(err, linstorClient.NotFoundError) {
+		return nil, nil
+	} else if err != nil {
+		return nil, fmt.Errorf("Could not find the resource definition: %w", err)
+	}
+
+	return &resourceDefinition, nil
+}
+
+// getResourceDefinitionName returns the Linstor resource definition name for a given `vol`.
+func (d *linstor) getResourceDefinitionName(vol Volume) string {
+	// TODO: set an unique ID on the name so we can later rename the resource at the
+	// Incus level, since Linstor does not support renaming
+	volName := vol.Name()
+	volType := vol.contentType
+
+	return fmt.Sprintf("%s-%s_%s", d.config[LinstorResourceGroupNameConfigKey], volName, volType)
+}
+
+// getLinstorDevPath return the device path for a given `vol`.
+func (d *linstor) getLinstorDevPath(vol Volume) (string, error) {
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return "", fmt.Errorf("Could not load Linstor client: %w", err)
```

Comment:
This adds a prefix that does not really add any information to an already prefixed error